src/NHSD.BuyingCatalogue.Solutions.Contracts/Queries/GetSuppliersByNameQuery.cs

```diff
@@ -6,11 +6,14 @@ namespace NHSD.BuyingCatalogue.Solutions.Contracts.Queries
 {
     public sealed class GetSuppliersByNameQuery : IRequest<IEnumerable<ISupplier>>
     {
-        public GetSuppliersByNameQuery(string name)
+        public GetSuppliersByNameQuery(string name, string solutionPublicationStatus)
```

Comment:
This would add complexity to the implementation and I'm struggling to see any benefit. I'm also not sure how this would prevent having to go to the database as we're not doing any caching as far as I'm aware. Do you mean it would avoid having to join to the `PublicationStatus` table? (Makes sense if so.)

In this case, I think changing the type of `SolutionPublicationStatus` to `PublishedStatus` would be better. This is a legitimate use of an `enum` – it is not being used to determine any behaviour, only to represent a set of values. Using the enum type will also mean that the out-of-box validation will work and will return a bad request response if an invalid enum value is presented.